client-src/default/index.js

```diff
@@ -27,12 +28,24 @@ const options = {
   useErrorOverlay: false,
   useProgress: false,
 };
-const socketUrl = createSocketUrl(__resourceQuery);
+const urlOptions = getUrlOptions(__resourceQuery);
+const socketUrl = createSocketUrl(urlOptions);
 
 self.addEventListener('beforeunload', () => {
   status.isUnloading = true;
 });
 
+if (urlOptions.query.logging) {
+  // this is needed because the HMR logger operate separately from
+  // dev server logger
+  const hotCtx = require.context('webpack/hot', false, /^\.\/log$/);
```

Comment:
Why do not use hot directly?

Reply:
Indeed, wasn't sure if it would break something but seems like it was some legacy check to previously support old Webpack versions.

Updated it with a classic import now